src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java

```diff
@@ -43,21 +70,83 @@ private MetadataDataAccessUtils() {}
 
   /** Generate an {@link AccessInfoModel} from a Snapshot */
   public static AccessInfoModel accessInfoFromSnapshot(final Snapshot snapshot) {
-    return makAccessInfo(
+    return makeAccessInfoBigQuery(
         snapshot.getName(),
         snapshot.getProjectResource().getGoogleProjectId(),
         snapshot.getTables());
   }
 
   /** Generate an {@link AccessInfoModel} from a Dataset */
-  public static AccessInfoModel accessInfoFromDataset(final Dataset dataset) {
-    return makAccessInfo(
-        BigQueryPdao.prefixName(dataset.getName()),
-        dataset.getProjectResource().getGoogleProjectId(),
-        dataset.getTables());
+  public AccessInfoModel accessInfoFromDataset(final Dataset dataset) {
+    CloudPlatformWrapper cloudPlatformWrapper =
+        CloudPlatformWrapper.of(dataset.getDatasetSummary().getStorageCloudPlatform());
+
+    if (cloudPlatformWrapper.isGcp()) {
+      return makeAccessInfoBigQuery(
+          BigQueryPdao.prefixName(dataset.getName()),
+          dataset.getProjectResource().getGoogleProjectId(),
+          dataset.getTables());
+    } else if (cloudPlatformWrapper.isAzure()) {
+      BillingProfileModel profileModel = dataset.getDatasetSummary().getDefaultBillingProfile();
+      AzureStorageAccountResource storageAccountResource =
+          resourceService.getStorageAccount(dataset, profileModel);
+      return makeAccessInfoAzure(
+          dataset.getName(), storageAccountResource, dataset.getTables(), profileModel);
+    } else {
+      throw new IllegalArgumentException("Unrecognized cloud platform");
+    }
+  }
+
+  private AccessInfoModel makeAccessInfoAzure(
+      final String datasetName,
+      final AzureStorageAccountResource storageAccountResource,
+      final List<? extends Table> tables,
+      final BillingProfileModel profileModel) {
+    AccessInfoModel accessInfoModel = new AccessInfoModel();
+
+    BlobContainerClientFactory targetDataClientFactory =
+        azureBlobStorePdao.getTargetDataClientFactory(
+            profileModel, storageAccountResource, ContainerType.METADATA, false);
+
+    // Given the sas token, rebuild a signed url
+    BlobSasTokenOptions options =
+        new BlobSasTokenOptions(
+            Duration.ofMinutes(15),
+            new BlobSasPermission().setReadPermission(true).setListPermission(true),
+            AzureSynapsePdao.class.getName());
+    String signedURL =
```

Comment:
I think we have a common signed URL method we should use here (it does the added thing of setting content disposition with the user name for tracing)